Sources/SwiftWarningControl/WarningControlRegions.swift

```diff
@@ -103,8 +103,16 @@ public struct WarningControlRegionTree {
   /// Root region representing top-level (file) scope
   private var rootRegionNode: WarningControlRegionNode
 
-  init(range: Range<AbsolutePosition>) {
-    rootRegionNode = WarningControlRegionNode(range: range)
+  /// Inheritance relationships between diagnostic group identifiers
+  /// Key: super-group, Values: sub-groups
+  let subGroupLinks: [DiagnosticGroupIdentifier: [DiagnosticGroupIdentifier]]
```

Comment:
You already call this inheritance in the doc comment and I think inheritance describes it better than `subGroupLinks`. It might be worth wrapping the dictionary in a struct so that we can have nice accessors that tell you what the sub and super-groups are. Something like the following

```swift
struct DiagnosticGroupInheritanceTree {
  private let subGroups: [DiagnosticGroupIdentifier: [DiagnosticGroupIdentifier]]

  /// Diagnostic groups that inherit from `group`.
  func subgroups(of group: DiagnosticGroupIdentifier) -> [DiagnosticGroupIdentifier] { subGroups[group] }
  ...
}
```

That way we might also add a static member `default` at some point if we want to, similar to how we have `standardOperators` in `SwiftOperators`.